### ISSUE TITLE
Service zone is not being rendered

### ISSUE THREAD

**Issue by matthenning** [Icinga/icingaweb2-module-director#1538]
## Expected Behavior
Selecting the `zone` will produce the matching `zone = "..."` field in the config.

## Current Behavior
It's visible in the resolved preview, but not the resulting file.

## Steps to Reproduce (for bugs)
Director config:
![grafik](https://user-images.githubusercontent.com/5699183/41659262-13de994a-7499-11e8-8ca0-3d67e3166f41.png)

Director preview (not visible here, but visible when toggling "Resolved"):
![grafik](https://user-images.githubusercontent.com/5699183/41659315-3a3ab06a-7499-11e8-8267-5a31e88d645c.png)

Resulting file, which is located under `/var/lib/icinga2/api/packages/director/<id>/zones.d/<satellite zone>` instead of the expected `zones.d/master`: 
```
template Service "***" {
    import "**"

    check_command = "check_elasticsearch_query"
    command_endpoint = null
    vars.crit_on_error = true
    vars.elasticsearch_hostname = "***"
    vars.elasticsearch_index = "filebeat-*"
    vars.elasticsearch_match = [
        "iq.sql_state.keyword ***",
        "fields.host_fqdn $host.name$"
    ]
    vars.elasticsearch_password = "***"
    vars.elasticsearch_port = "9200"
    vars.elasticsearch_time_diff = "now-15m"
    vars.elasticsearch_username = "***"
}
```

## Your Environment

* Director version (System - About): 1.4.3
* Icinga Web 2 version and modules (System - About): 2.5.3
* Icinga 2 version (`icinga2 --version`): 2.8.4
* Operating System and version: Debian 9.4
* Webserver, PHP versions: nginx, PHP 7.0

**Comment by matthenning** [comment]
Thank you but I already verified that.
You can see the resulting file in my original post.
Also the resulting service config is not rendered to zones.d/master/ but the the directory of the satellite zone.

**Comment by swizzly** [comment]
I just checked, and in my case the service gets rendered in the master zone if the Cluster Zone is "master". But, in this case the Problem is, that the given service is not getting executed anymore from hosts in any other satellite zone.
I have director fc58e50.

**Comment by swizzly** [comment]
@matthenning The Problem still exists with the current Director version (44af7b3). @Thomas-Gelf helped me with the following code, that solves this Problem, hope that it gets to the director master soon. Try out:

```
# diff -u  /tmp/icingaweb2-module-director-master/library/Director/Objects/IcingaService.php /usr/share/icingaweb2/modules/director/library/Director/Objects/IcingaService.php
--- /tmp/icingaweb2-module-director-master/library/Director/Objects/IcingaService.php	2018-11-27 19:02:39.000000000 +0100
+++ /usr/share/icingaweb2/modules/director/library/Director/Objects/IcingaService.php	2018-11-12 14:02:29.506015600 +0100
@@ -418,6 +418,18 @@
      */
     protected function renderObjectHeader()
     {
+        if ($this->isApplyRule()) {
+            $zone = $this->getRenderingZone();
+            if ($zone === 'director-global') {
+                $zone = '';
+            } else {
+                $zone = c::renderKeyValue('zone', c::renderString($this->getRenderingZone()));
+                // TODO: add ignore where host.zone != $zone???
+            }
+        } else {
+            $zone = '';
+        }
+
         if ($this->isApplyRule()
             && !$this->hasBeenAssignedToHostTemplate()
             && $this->get('apply_for') !== null
@@ -438,10 +450,11 @@
                 $this->getType(),
                 $name,
                 $this->get('apply_for')
-            ) . $extraName;
+            ) . $extraName . $zone;
         }
 
-        return parent::renderObjectHeader();
+
+        return parent::renderObjectHeader() . $zone;
     }
 
     /**

```

**Comment by Thomas-Gelf** [comment]
A change similar to the shown one has been merged into v1.7. Please let me know when still facing this issue.
